hsrc_lib/Database/LMDB/Raw.hsc

```diff
@@ -1305,6 +1307,16 @@ mdb_cursor_put' wf crs key val =
     r_cursor_put rc
 {-# INLINE mdb_cursor_put' #-}
 
+-- | Like 'mdb_cursor_put', but with direct control of how pointers to values
+-- are allocated, whether an argument is a nullPtr, and so on.
+mdb_cursor_put_ptr :: MDB_WriteFlags -> MDB_cursor -> Ptr MDB_val -> Ptr MDB_val -> IO Bool
+mdb_cursor_put_ptr wf crs pKey pVal = _mdb_cursor_put (_crs_ptr crs) pKey pVal wf >>= r_cursor_put
```

Comment:
This function seems more general than `mdb_cursor_put`, why not implement it in terms of this new function?